### PR TITLE
Drop support for deprecated attribute `gl3d.cameraposition` (use `gl3d.camera` instead)

### LINKED ISSUE #7196: drop deprecated gl3d `cameraposition` in v3
https://github.com/plotly/plotly.js/blob/206d117e01a79ff63b5a8f5f76d97e2b095e8ac5/src/plots/gl3d/layout/layout_attributes.js#L163-L169

### PR DESCRIPTION
**Pull request by emilykl**
Closes #7196 

- Drop support for deprecated attribute `gl3d.cameraposition` (use `gl3d.camera` instead)
- Update mocks

### CODE REVIEW

**Comment by emilykl** [comment]
@archmoj @LiamConnors Do we have docs anywhere explaining how to convert from the old `cameraposition` to the new `camera` format? If not, should we? 

The conversion can be figured out by looking at [this function](https://github.com/plotly/plotly.js/pull/7217/files#diff-39ff2a6111d28f1f7183c2927e07895b5adb4f977352fb762040d63628af4a68).

File: test/image/mocks/gl3d_opacity-surface.json [hunk 1]
```diff
@@ -1540,14 +1540,23 @@
...
-    [
-     -0.8326258918343652, -0.31013201281074904, -0.14770225080852764,
-     0.4344379172943774
-    ],
-    [-0.09586835824120499, -0.07248137902934104, 0.013764797418843955],
-    1.9263438581634802
-   ],
+   "camera": {
+    "eye": {
+     "x": 0.8970225259545955,
```
Comment (gvwilson) [inline test/image/mocks/gl3d_opacity-surface.json:1545]: aaaand we just *know* these numbers? :-)

**Review by gvwilson — APPROVED** [review]

File: test/image/mocks/gl3d_opacity-surface.json (hunk 1 above)
Comment (emilykl) [reply test/image/mocks/gl3d_opacity-surface.json:1545]: I used [the existing conversion function](https://github.com/plotly/plotly.js/pull/7217/files#diff-39ff2a6111d28f1f7183c2927e07895b5adb4f977352fb762040d63628af4a68) before deleting it. :-) 

But I'll admit I don't actually know what the numbers in the old format mean. The meaning of the new format is documented [here](https://plotly.com/javascript/reference/#layout-scene-camera).

**Comment by LiamConnors** [comment]
> @archmoj @LiamConnors Do we have docs anywhere explaining how to convert from the old `cameraposition` to the new `camera` format? If not, should we?
> 
> The conversion can be figured out by looking at [this function](https://github.com/plotly/plotly.js/pull/7217/files#diff-39ff2a6111d28f1f7183c2927e07895b5adb4f977352fb762040d63628af4a68).

Do you know when was this originally deprecated?

> @archmoj @LiamConnors Do we have docs anywhere explaining how to convert from the old `cameraposition` to the new `camera` format? If not, should we?
> 
> The conversion can be figured out by looking at [this function](https://github.com/plotly/plotly.js/pull/7217/files#diff-39ff2a6111d28f1f7183c2927e07895b5adb4f977352fb762040d63628af4a68).

There's nothing currently in the docs. I'll write up a page that lists the deprecation removals and how to migrate.

**Comment by emilykl** [comment]
> I'll write up a page that lists the deprecation removals and how to migrate.

That would be fantastic -- thanks @LiamConnors !

**Comment by LiamConnors** [comment]
@emilykl do you know which traces this was available on? I'm having trouble identifying exactly which ones.

**Comment by emilykl** [comment]
@LiamConnors I believe it's just [3d surface plots](https://plotly.com/python/3d-surface-plots/) (is that correct @archmoj?)

**Pull request merged**